### PR TITLE
Allow specifying the group while whitelisting Slack workflows

### PR DESCRIPTION
**Pull request by lasryaric**
## Description

Allow specifying the group while whitelisting Slack workflows.

Last part of https://github.com/dust-tt/dust/issues/6460

Screenshot 2024-08-26 at 17 04 04

## Deploy Plan

- deploy connectors
- Run migration connectors migration id 12 from prodbox.
- run the backfill script from front

### CODE REVIEW

File: front/migrations/20240826_backfill_slack_whitelisted_bots.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import { GroupResource } from "@app/lib/resources/group_resource";
+import { makeSId } from "@app/lib/resources/string_ids";
+import { makeScript } from "@app/scripts/helpers";
+
+makeScript({}, async ({ execute }, logger) => {
+  const connectorDB = new Sequelize(
+    EnvironmentConfig.getEnvVariable("CONNECTORS_DATABASE_URI")
+  );
+  const bots: { id: number; workspaceid: string }[] = await connectorDB.query(
+    `SELECT swb.id as id, c."workspaceId" as workspaceid FROM slack_bot_whitelist swb INNER JOIN connectors c ON swb."connectorId" = c.id WHERE c.type = 'slack'`,
```
Comment (flvndvd) [inline front/migrations/20240826_backfill_slack_whitelisted_bots.ts:15]:
```suggestion
    `SELECT sbw.id as id, c."workspaceId" as workspaceId FROM slack_bot_whitelist sbw INNER JOIN connectors c ON sbw."connectorId" = c.id WHERE c.type = 'slack'`,
```

File: front/pages/poke/[wId]/data_sources/[name]/index.tsx [hunk 2]
```diff
@@ -226,6 +231,16 @@ export const getServerSideProps = withSuperUserAuthRequirements<{
...
   }
 
+  // Getting groups that a Slack bot/workflow can be assigned to
+  const authForSlackBot = await Authenticator.internalAdminForWorkspace(
+    owner.sId
+  );
+  const groupsForSlackBot = (
+    await GroupResource.listWorkspaceGroups(authForSlackBot)
+  )
+    .filter((g) => g.kind !== "system")
```
Comment (flvndvd) [inline front/pages/poke/[wId]/data_sources/[name]/index.tsx:241]: Let's add a `isSystem` method on the `GroupResource`.

**Review by flvndvd — APPROVED** [review]
LGTM, left two small nits but good to merge once addressed.

File: front/pages/poke/[wId]/data_sources/[name]/index.tsx (hunk 2 above)
Comment (lasryaric) [reply front/pages/poke/[wId]/data_sources/[name]/index.tsx]: 👍

**Pull request merged**